src/components/KuadrantDNSPolicyCreatePage.tsx

```diff
@@ -0,0 +1,327 @@
+import * as React from 'react';
+import Helmet from 'react-helmet';
+import yaml from 'js-yaml';
+import {
+  PageSection,
+  Title,
+  TextInput,
+  FormGroup,
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+  Form,
+  Radio,
+  Button,
+  ExpandableSection,
+  ButtonVariant,
+  Modal,
+  ActionGroup,
+} from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import './kuadrant.css';
+import { k8sCreate, ResourceYAMLEditor } from '@openshift-console/dynamic-plugin-sdk';
+import { useHistory } from 'react-router-dom';
+import { LoadBalancing, HealthCheck, DNSPolicy, WeightedCustom } from './dnspolicy/types'
+import LoadBalancingField from './dnspolicy/LoadBalancingField';
+import HealthCheckField from './dnspolicy/HealthCheckField';
+import getModelFromResource from '../utils/getModelFromResource';
+import { Gateway } from './gateway/types';
+import GatewaySelect from './gateway/GatewaySelect';
+import { ModalHeader, ModalBody, ModalFooter } from '@patternfly/react-core/next';
+import NamespaceSelect from './namespace/NamespaceSelect';
+import { removeUndefinedFields, convertMatchLabelsArrayToObject, convertMatchLabelsObjectToArray } from '../utils/dnsPolicy';
+
+const KuadrantDNSPolicyCreatePage: React.FC = () => {
+  const { t } = useTranslation('plugin__console-plugin-template');
+  const [createView, setCreateView] = React.useState<'form' | 'yaml'>('form');
+  const [policy, setPolicy] = React.useState('');
+  const [selectedNamespace, setSelectedNamespace] = React.useState('');
+  const [selectedGateway, setSelectedGateway] = React.useState<Gateway>({ name: '', namespace: '' });
+  const [routingStrategy, setRoutingStrategy] = React.useState<'simple' | 'loadbalanced'>('simple');
+  const [loadBalancing, setLoadBalancing] = React.useState<LoadBalancing>({
+    geo: { defaultGeo: '' },
+    weighted: { defaultWeight: 0 }
+  });
+  const [healthCheck, setHealthCheck] = React.useState<HealthCheck>({
+    endpoint: '',
+    failureThreshold: null,
+    port: null,
+    protocol: 'HTTP',
+  });
+
+  // Initialize the YAML resource object based on form state
+  const [yamlResource, setYamlResource] = React.useState(() => {
+    return removeUndefinedFields({
+      apiVersion: 'kuadrant.io/v1alpha1',
+      kind: 'DNSPolicy',
+      metadata: {
+        name: policy,
+        namespace: selectedNamespace,
+      },
+      spec: {
+        routingStrategy,
+        targetRef: {
+          group: 'gateway.networking.k8s.io',
+          kind: 'Gateway',
+          name: selectedGateway.name,
+          namespace: selectedGateway.namespace,
+        },
+        loadBalancing: routingStrategy === 'loadbalanced' ? loadBalancing : undefined,
+        healthCheck: healthCheck.endpoint ? healthCheck : undefined,
+      },
+    });
+  });
+
+  const history = useHistory();
+
+  React.useEffect(() => {
+    const updatedYamlResource = {
+      apiVersion: 'kuadrant.io/v1alpha1',
```

Comment:
small thing, but I guess this applies to all of these views - should we have a helper to fetch the latest apiVersions for our resources? saves us having to deal with drift here (well, for the most part). I guess ditching the apiVersion would do the same.